shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -495,13 +497,58 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
     lifecycle.handleLifecycleEvent(Lifecycle.Event.ON_CREATE);
 
+    registerOnBackInvokedCallback();
+
     configureWindowForTransparency();
 
     setContentView(createFlutterView());
 
     configureStatusBarForFullscreenFlutterExperience();
   }
 
+  /**
+   * Registers the callback with OnBackInvokedDispatcher to capture back navigation gestures and
+   * pass them to the framework.
+   *
+   * <p>This replaces the deprecated onBackPressed method override in order to support API 33's
+   * predictive back navigation feature.
+   *
+   * <p>The callback must be unregistered in order to prevent unpredictable behavior once outside
+   * the Flutter app.
+   */
+  @VisibleForTesting
+  public void registerOnBackInvokedCallback() {
+    if (Build.VERSION.SDK_INT >= 33) {
+      getOnBackInvokedDispatcher()
+          .registerOnBackInvokedCallback(
+              OnBackInvokedDispatcher.PRIORITY_DEFAULT, onBackInvokedCallback);
+    }
+  }
+
+  /**
+   * Unregisters the callback from OnBackInvokedDispatcher.
+   *
+   * <p>This should be called when the activity is no longer in use to prevent unpredictable
+   * behavior such as being stuck and unable to press back.
+   */
+  @VisibleForTesting
+  public void unregisterOnBackInvokedCallback() {
+    if (Build.VERSION.SDK_INT >= 33) {
+      getOnBackInvokedDispatcher().unregisterOnBackInvokedCallback(onBackInvokedCallback);
+    }
+  }
+
+  private final OnBackInvokedCallback onBackInvokedCallback =
+      Build.VERSION.SDK_INT >= 33
+          ? new OnBackInvokedCallback() {
+            @SuppressWarnings("Override")
```

Comment:
Please add a TODO for removing this, explaining when it can be removed (probably best to file a Flutter issue tracking it to reference in the TODO, and reference the internal bug in that issue along with any other useful context).

Reply:
Done. Filed https://github.com/flutter/flutter/issues/111295 and added TODO.